docs/content/preview/api/ysql/the-sql-language/statements/ddl_alter_table.md

```diff
@@ -198,6 +198,49 @@ It quietly succeeds. Now `\d children` shows that the foreign key constraint `ch
 
 Add the specified constraint to the table.
 
+#### [*alter_column_type*]
+
+Change the type of an existing column. Depending if data on disk is required to be changed, different semantics apply.
+
+Indexes and simple table constraints involving the column will be automatically converted to use the new column type by reparsing the originally supplied expression. 
+
+The optional `COLLATE` clause specifies a collation for the new column; if omitted, the collation is the default for the new column type. 
+
+The optional `USING` clause specifies how to compute the new column value from the old; if omitted, the default conversion is the same as an assignment cast from old data type to new. 
+
+A `USING` clause must be provided if there is no implicit or assignment cast from old to new type.
+
+##### Online alter type
+ 
+If the change doesn't require data on disk to change, these semantics apply:
+
+- The action is online and concurrent DMLs to the table can go through.
+
+Example:
+
+```sql
+CREATE TABLE test (id BIGSERIAL PRIMARY KEY, a VARCHAR(50));
+ALTER TABLE test ALTER COLUMN a TYPE VARCHAR(51);
+```
+
+##### Offline alter type
```

Comment:
online/offline is a bad way to put it.  It isn't particularly offline in that it doesn't block operations.  You even say "should not be performed" rather than "cannot be performed".

It should instead be branched at whether table rewrite is required.

Reply:
I realize that I'm not qualified to review this Doc PR. Please remove me from the list.

Reply:
adding to @jasonyb 's comment perhaps a better way to describe it is that the table rewrite scenario is unsafe (concurrent DMLs may result in data inconsistency etc)